### PR TITLE
Updated tooling to use Golang 1.19.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Updating our tooling to use Golang 1.19. This enables us to use Golang's generics to make some of our helper functions more robust and simplify the code.

###### Change Log  

- Switched Mariner's toolkit to Golang 1.19.
- Switched `sliceutils` helper functions to use generics.
- Updated GitHub workflows to use Golang 1.19.
- Updated remaining GitHub workflows to use `ubuntu-latest`.

###### Does this affect the toolchain?  

No.

###### Test Methodology

- [Full pipeline build](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=385313&view=results).

### CODE REVIEW

**Comment by trungams** [comment]
There was [this PR](https://github.com/microsoft/CBL-Mariner/pull/5630) to update the recommended version to 1.19. Would there be any problem bumping golang to 1.19 or even 1.20 instead? I think 1.18 has also reached EOL

**Review by oliviacrain — APPROVED** [review]

**Comment by PawelWMS** [comment]
> > There was [this PR](https://github.com/microsoft/CBL-Mariner/pull/5630) to update the recommended version to 1.19. Would there be any problem bumping golang to 1.19 or even 1.20 instead? I think 1.18 has also reached EOL
> 
> I agree we should eventually land on only supporting non-EOL versions of Go, but I'd personally prefer we handle one minor version jump at a time.

I missed the fact we're recommending 1.19 now as the supported version. In that case, I actually think we can try doing the jump directly to 1.19 now. I'll update the PR and re-run the full build.

File: .github/workflows/quickstart_1.0.yml [hunk 1]
```diff
@@ -12,18 +12,18 @@ on:
...
+    runs-on: ubuntu-latest
 
     steps:
     - name: Checkout
       uses: actions/checkout@v3
       with:
         ref: '1.0-stable'
 
-    - name: Set up Go 1.17
+    - name: Set up Go 1.18
```
Comment (trungams) [inline .github/workflows/quickstart_1.0.yml:23]: the name fields in these files need updating as well

**Review by trungams — CHANGES_REQUESTED** [review]

File: .github/workflows/quickstart_1.0.yml (hunk 1 above)
Comment (PawelWMS) [reply .github/workflows/quickstart_1.0.yml:23]: It should be fine now, thanks.

Comment (trungams) [reply .github/workflows/quickstart_1.0.yml:23]: wait shouldn't it be 1.19 instead of 1.18?

Comment (PawelWMS) [reply .github/workflows/quickstart_1.0.yml]: Oh, sorry, I missed that one, I focused on the `go-version` fields.:/ It should be fine now.

**Review by trungams — APPROVED** [review]

**Review by oliviacrain — APPROVED** [review]

**Pull request merged**